includes/class-amp-story-post-type.php

```diff
@@ -174,2197 +102,179 @@ public static function register() {
 		}
 
-		$style_attr_values = [];
-
-		// Replace inline styles with temporary data-temp-style-hash attribute before KSES...
-		add_filter(
-			'content_save_pre',
-			static function ( $post_content ) use ( &$style_attr_values ) {
-				$post_content = preg_replace_callback(
-					'|(?P<before><\w+(?:-\w+)*\s[^>]*?)style=\\\"(?P<styles>[^"]*)\\\"(?P<after>([^>]+?)*>)|', // Extra slashes appear here because $post_content is pre-slashed..
-					static function ( $matches ) use ( &$style_attr_values ) {
-						$hash                       = md5( $matches['styles'] );
-						$style_attr_values[ $hash ] = self::safecss_filter_attr( wp_unslash( $matches['styles'] ) );
-
-						// Replaces the complete style attribute value with its hashed version.
-						return $matches['before'] . sprintf( ' data-temp-style-hash="%s" ', $hash ) . $matches['after'];
-					},
-					$post_content
-				);
+		/**
+		 * Filter list of allowed video mime types.
+		 *
+		 * This can be used to add additionally supported formats, for example by plugins
+		 * that do video transcoding.
+		 *
+		 * @since 1.3
+		 *
+		 * @param array Allowed video mime types.
+		 */
+		$allowed_video_mime_types = apply_filters( 'amp_story_allowed_video_types', [ 'video/mp4' ] );
 
-				return $post_content;
-			},
-			0
-		);
+		// If `$allowed_video_mime_types` doesn't have valid data or is empty add default supported type.
+		if ( ! is_array( $allowed_video_mime_types ) || empty( $allowed_video_mime_types ) ) {
+			$allowed_video_mime_types = [ 'video/mp4' ];
+		}
 
-		// ...And bring it back afterwards.
-		add_filter(
-			'content_save_pre',
-			static function ( $post_content ) use ( &$style_attr_values ) {
-				// Replaces hashed style attribute value with the original value again.
-				return preg_replace_callback(
-					'/ data-temp-style-hash=\\\"(?P<hash>[0-9a-f]+)\\\"/',
-					static function ( $matches ) use ( $style_attr_values ) {
-						return isset( $style_attr_values[ $matches['hash'] ] ) ? sprintf( ' style="%s"', esc_attr( wp_slash( $style_attr_values[ $matches['hash'] ] ) ) ) : '';
-					},
-					$post_content
-				);
-			},
-			20
-		);
+		// Only add currently supported mime types.
+		$allowed_video_mime_types = array_values( array_intersect( $allowed_video_mime_types, wp_get_mime_types() ) );
 
-		return $response;
-	}
+		/**
+		 * Filters the list of allowed post types for use in page attachments.
+		 *
+		 * @since 1.3
+		 *
+		 * @param array Allowed post types.
+		 */
+		$page_attachment_post_types = apply_filters( 'amp_story_allowed_page_attachment_post_types', [ 'page', 'post' ] );
+		$post_types                 = [];
+		foreach ( $page_attachment_post_types as $post_type ) {
+			$post_type_object = get_post_type_object( $post_type );
 
-	/**
-	 * Filter the allowed tags for KSES to allow for amp-story children.
-	 *
-	 * @param array $allowed_tags Allowed tags.
-	 * @return array Allowed tags.
-	 */
-	public static function filter_kses_allowed_html( $allowed_tags ) {
```

Comment:
This part we'd still need so that it wouldn't get filtered out from the FE from `post_content`.

Reply:
This should be in another PR / issue. Preview and saving content should be discussed in #3838

Reply:
OK, will address it there instead. 👍